real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/ui/LoggingPanel.java

```diff
@@ -0,0 +1,122 @@
+package com.roboclub.robobuggy.ui;
+
+import java.awt.Color;
+import java.awt.Font;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.sql.NClob;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+import javax.swing.JButton;
+import javax.swing.JFormattedTextField;
+import javax.swing.JLabel;
+import javax.swing.SwingConstants;
+import javax.swing.Timer;
+
+import com.roboclub.robobuggy.logging.RobotLogger;
+import com.roboclub.robobuggy.main.RobobuggyMessageLevel;
+import com.roboclub.robobuggy.main.RobobuggyConfigFile;
+import com.roboclub.robobuggy.main.RobobuggyLogicException;
+import com.roboclub.robobuggy.messages.GuiLoggingButtonMessage;
+import com.roboclub.robobuggy.ros.NodeChannel;
+import com.roboclub.robobuggy.ros.Publisher;
+
+public class LoggingPanel extends RoboBuggyGUIContainer{
+	private static JButton play_btn;
+	private JFormattedTextField time_lbl;
+	private static final int TIME_ZONE_OFFSET = 18000000;//5 hours
+	private static Date startTime;
+	private static Timer timer;
+	Publisher logging_button_pub;
+	private JLabel filename_lbl;
+	//to make logging panel accessible to button callbacks
+	private LoggingPanel thisLoggingPanel = this;
+	
+
+	public LoggingPanel(){
+		name = "LoggingPanel";
+		logging_button_pub = new Publisher(NodeChannel.GUI_LOGGING_BUTTON.getMsgPath());
+
+		timer = new Timer(10, new timerHandler());// updates every .01 seconds
+		timer.setDelay(100);
+		timer.setRepeats(true); // timer needs to be setup before startpause_btn
+
+		//should be the time that we start the sytem at 
+		startTime = new Date();
+		
+		play_btn = new JButton("START");
+		play_btn.setFont(new Font("serif", Font.PLAIN, 70));
+		play_btn.addActionListener(new PlayButtonHandler());
+		play_btn.setEnabled(true);
+		play_btn.setBackground(Color.BLUE);
+	
+		filename_lbl = new JLabel("File: ",
+				SwingConstants.CENTER);
+		filename_lbl.setFont(new Font("sanserif", Font.PLAIN, 15));
+
+		time_lbl = new JFormattedTextField(new SimpleDateFormat("HH:mm:ss.S"));
+		time_lbl.setHorizontalAlignment(SwingConstants.CENTER);
+		time_lbl.setFont(new Font("sanserif", Font.PLAIN, 50));
+		time_lbl.setEditable(false);
+		time_lbl.setColumns(7);
+		time_lbl.setValue(startTime);
+
+		this.addComponent(play_btn, 0, 0, 1.0, .25);
+		this.addComponent(filename_lbl, 0, .25, 1.0, .25);
+		this.addComponent(time_lbl, 0, .5, 1, .5);
+	}
+
+	public void setFileName(String fileName) {
+		filename_lbl.setText("File: " + fileName);
+	}
+
+	private class PlayButtonHandler implements ActionListener {
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Each class should declare at least one constructor](https://www.codacy.com/app/Mr-Trevor-Decker/RoboBuggy/file/1814872849/issues/source?bid=2938665&fileBranchId=3059924#l74)